File: src/main/java/me/muksc/tacztweaks/mixininterface/feature/raytracer/RayTracingBullet.java
package me.muksc.tacztweaks.mixininterface.feature.raytracer;

import com.tacz.guns.entity.EntityKineticBullet;
import net.minecraft.world.phys.Vec3;

public interface RayTracingBullet {
    static RayTracingBullet of(EntityKineticBullet instance) {
        return (RayTracingBullet) instance;
    }

    Vec3 tacztweaks$getCurrentHitPosition();

    void tacztweaks$setCurrentHitPosition(Vec3 position);
}

File: src/main/kotlin/me/muksc/tacztweaks/mixininterop/RayTracingBullet.kt
package me.muksc.tacztweaks.mixininterop

import me.muksc.tacztweaks.mixininterface.feature.raytracer.RayTracingBullet
import net.minecraft.world.phys.Vec3

inline var RayTracingBullet.currentHitPosition: Vec3
    get() = `tacztweaks$getCurrentHitPosition`()
    set(value) = `tacztweaks$setCurrentHitPosition`(value)